packages/mui-base/src/ListboxUnstyled/defaultListboxReducer.ts

```diff
@@ -228,6 +228,74 @@ function handleBlur<TOption>(state: ListboxState<TOption>): ListboxState<TOption
   };
 }
 
+export const textCriteriaMatches = <TOption>(
+  nextFocus: TOption,
+  textCriteriaStr: string,
+  stringifyOption: (option: TOption) => string | undefined,
+) => {
+  const text = stringifyOption(nextFocus)?.trim().toLowerCase();
+
+  if (!text || text.length === 0) {
+    // Make option not navigable if stringification fails or results in empty string.
+    return false;
+  }
+
+  return text.indexOf(textCriteriaStr) === 0;
+};
+
+function handleTextNavigation<TOption>(
+  state: ListboxState<TOption>,
+  textCriteriaStr: string,
```

Comment:
We try not to use abbreviations. In this case, `searchString` (consistent with https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith#syntax) or `prefix` would be better (the same applies to `textCriteriaMatches` parameters above)